src/main/java/com/warrenstrange/googleauth/GoogleAuthenticator.java

```diff
@@ -166,10 +168,44 @@ public GoogleAuthenticator(GoogleAuthenticatorConfig config)
         }
 
         this.config = config;
+
+        this.secureRandom = new ReseedingSecureRandom(
+                getRandomNumberAlgorithm(),
+                getRandomNumberAlgorithmProvider()
+        );
+    }
+
+    public GoogleAuthenticator(final String randomNumberAlgorithm, final String randomNumberAlgorithmProvider)
+    {
+        this(new GoogleAuthenticatorConfig(), randomNumberAlgorithm, randomNumberAlgorithmProvider);
+
+    }
+
+    public GoogleAuthenticator(GoogleAuthenticatorConfig config, final String randomNumberAlgorithm, final String randomNumberAlgorithmProvider)
```

Comment:
Did you consider adding `randomNumberAlgorithm` and `randomNumberAlgorithmProvider` to the `GoogleAuthenticatorConfig`? I'm not sure why these particular properties were only exposed via system properties. But if we're going to expose another way to configure them, the config might be better than an overloaded constructor.

Reply:
@emcrisostomo I thought about adding the change that @jdullum is referring to. I think that that would be better than the current change I made. My hesitation is that `DEFAULT_RANDOM_NUMBER_ALGORITHM` and `DEFAULT_RANDOM_NUMBER_ALGORITHM_PROVIDER` were already defined in GoogleAuthenticator. Adding the algorithm and provider to the config would mean moving the default algorithm and provider to the config, which would be a refactor. Since this is your repo and not ours I want to follow existing patterns and be minimally invasive. That being said, what are your thoughts on the change described? I'd be happy to make that change or stick to the existing pattern.

Reply:
I agree with you @jdullum , @EricChristensen , I agree with this change, I'm going to merge this anyway, and if you're willing to provide a PR I'll be happy to consider it.